feat(ui/calibration): add state machine and instructions

zacowan opened the pull request:
This branch adds the state machine for calibration as well as a placeholder for instructions.

Update: this branch now also swaps our formatter to black (from autopep8) and adds some updates to pylint.

zacowan committed:
feat(ui/calibration): add state machine

zacowan committed:
feat(ui/calibration): add close to state machine

zacowan committed:
chore(config): add github action

zacowan committed:
chore(config): swap to black, update action

zacowan committed:
chore(actions): add black

zacowan committed:
chore(format): fix formatting, linting

zacowan committed:
chore(action): remove action

Scoobydoo181 commented:
The changes to the ui code look good, but why so many changes to detectEyes.py? Some of them are arbitrary like changing double quotes to single quotes, but a lot of them significantly worsen readability.

Also, the .vscode config folder should probabally be in the .gitignore since our code should be editor-independent.

zacowan committed:
chore(config): remove .vscode

zacowan committed:
chore(ignore): add .vscode

zacowan committed:
chore(detectEyes): revert to main

zacowan commented:
@Scoobydoo181 thanks for the feedback 😄 

For the changes to `detectEyes.py`, those were mostly just the autoformatting from `black` and some minor tweaks to fix `pylint` errors whenever I was experimenting with adding a GitHub action for lining and style checking. I've since removed the GitHub action, because `pylint` is not comprehensive enough to work with our packages (specifically, `pyside6` and `cv2`). Anyways, those changes were leftover and I've reverted `detectEyes.py` back to the way it was on our main branch.

Also agree with your `.gitignore` suggestion, I've added that in!
